lib/facility.rb

```diff
@@ -1,14 +1,60 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless @services.include?("Vehicle Registration")
+    vehicle.assign_registration_date
+    vehicle.assign_plate_type
+    registration_fee(vehicle)
+    @registered_vehicles << vehicle
+    @registered_vehicles
+  end
 end
+
+  def registration_fee(vehicle)
```

Comment:
Remember that method names should be verbs so maybe something like `calculate_reg_fee`